### DISCUSSION TITLE
자바스크립트 프로그램의 파싱/컴파일 단계에서 AST는 어떤 역할을 하나요?

### DISCUSSION THREAD

**Discussion by lledellebell** [diving-javascript/you-dont-know-js-yet#59] [1. 스코프] [answered]
<details>
<summary>관련 챕터</summary>
[Part2] 1.4 자바스크립트의 두 단계 (Required: Two Phases)
</details>

**Comment by wogus216** [comment, marked as answer]
<details>
<summary>답변</summary>


AST(추상 구문 트리)는 JS 프로그램의 파싱 및 컴파일 단계에서 **소스 코드를 구조적으로 분석하고 최적화하며, 최종 실행 코드를 생성하는 데 필수적인 역할**을 합니다.
AST를 통해 컴파일러는 코드의 구문을 이해하고, 최적화 및 변환을 수행하며, 디버깅 및 코드 분석 도구는 개발자에게 유용한 정보를 제공
</details>
